### PR TITLE
SDK Module Runner for local method execution

### PR DESCRIPTION
**Pull request by rsutormin**

### CODE REVIEW

**Comment by rsutormin** [comment]
@MrCreosote and @msneddon: I will add some bugfixes later but you can actually review my code already.

**Comment by MrCreosote** [comment]
Feedback so far:
Is there a way to provide a mount point so you can run methods that require file input or output (ReadsUtils for example)? I'm not seeing any way to do that.

runDir should probably be an argument so that the user can specify where all the files should go.

There aren't any tests.

Standard comment about unbracketed blocks

**Comment by MrCreosote** [comment]
The PR I just sent you means the the callback server will use a logger. I'm planning to sync the njsw execution engine files -> kb_sdk once that PR is merged.

File: src/java/us/kbase/mobu/runner/ModuleRunner.java [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+        rpc.put("context", new LinkedHashMap<String, Object>());
+        UObject.getMapper().writeValue(new File(workDir, "input.json"), rpc);
+        ////////////////////////////////// Starting callback service //////////////////////////////
+        int callbackPort = NetUtils.findFreePort();
+        URL callbackUrl = CallbackServer.getCallbackUrl(callbackPort);
+        Server jettyServer = null;
+        if (callbackUrl != null) {
+            if( System.getProperty("os.name").startsWith("Windows") ) {
+                JsonServerSyslog.setStaticUseSyslog(false);
+                JsonServerSyslog.setStaticMlogFile("callback.log");
```
Comment (MrCreosote) [inline src/java/us/kbase/mobu/runner/ModuleRunner.java]: this should go in the same directory as all the other files.

File: src/java/us/kbase/mobu/runner/ModuleRunner.java [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+                        @Override
+                        public void logNextLine(String line, boolean isError) {
+                            //do nothing, SDK callback server doesn't use a logger
+                        }
+                    }).build();
+            ModuleRunVersion runver = new ModuleRunVersion(
+                    new URL("https://fakefakefakefakefake.com"),
+                    new ModuleMethod("use_set_provenance.to_set_provenance_for_tests"),
+                    "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", "0.0.0", "dev");
+            JsonServerServlet catalogSrv = new SDKCallbackServer(
```
Comment (MrCreosote) [inline src/java/us/kbase/mobu/runner/ModuleRunner.java]: Why not fill this in correctly? The information is available from the catalog call above.

File: src/java/us/kbase/mobu/runner/ModuleRunner.java [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+                    auth, cfg, runver, new ArrayList<UObject>(),
+                    new ArrayList<String>());
```
Comment (MrCreosote) [inline src/java/us/kbase/mobu/runner/ModuleRunner.java]: Same for the input parameters. Also, it'd be good to allow specifying a list of object references to set in the provenance in the CLI

**Comment by MrCreosote** [comment]
As of the last commit the current set of tests pass.

File: src/java/us/kbase/mobu/runner/ModuleRunner.java [hunk 4]
```diff
@@ -49,26 +54,28 @@
...
+        if (sdkHome == null) {
+            sdkHome = System.getenv(ModuleBuilder.GLOBAL_SDK_HOME_ENV_VAR);
+            if (sdkHome == null)
+                throw new IllegalStateException("Path to kb-sdk home folder should be set either" +
+                		" in command line (-h) or in " + ModuleBuilder.GLOBAL_SDK_HOME_ENV_VAR + 
+                		" system environment variable");
+        }
+        File sdkHomeDir = new File(sdkHome);
+        if (!sdkHomeDir.exists())
+            throw new IllegalStateException("Directory " + sdkHome + " doesn't exist");
```
Comment (MrCreosote) [inline src/java/us/kbase/mobu/runner/ModuleRunner.java]: Why not just make it if it doesn't exist?

File: src/java/us/kbase/mobu/runner/ModuleRunner.java [hunk 5]
```diff
@@ -211,13 +240,19 @@ public void logNextLine(String line, boolean isError) {
...
                         }
                     }).build();
-            ModuleRunVersion runver = new ModuleRunVersion(
-                    new URL("https://fakefakefakefakefake.com"),
-                    new ModuleMethod("use_set_provenance.to_set_provenance_for_tests"),
-                    "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", "0.0.0", "dev");
+            Set<String> releaseTags = new TreeSet<String>(mv.getReleaseTags());
+            System.out.println("Release tags: " + releaseTags);
+            String requestedRelease = null;
+            final ModuleRunVersion runver = new ModuleRunVersion(
```
Comment (MrCreosote) [inline src/java/us/kbase/mobu/runner/ModuleRunner.java]: Why is requested release null? It's supplied on the command line (well, that or a git commit).

**Comment by MrCreosote** [comment]
Although there are no tests, we agreed to merge since people need this. I strongly advise adding tests later.

**Pull request merged**